Update vehicle_thinkcity.cpp

MylesTwete opened the pull request:
TPMS Status (0x345) Case added.  This addition uses Standard Metrics for TPMS Alerts and Warnings to allow TPMS Sensor Status data to be reflected in the TPMS GUI of the vehicle App as well as in the notifications.  These changes were included in a local build, tested on 2 THINK City vehicles and functionality verified.   
![Screenshot_20251119_123257_Open Vehicle](https://github.com/user-attachments/assets/78f011a4-8eee-48fd-b6e0-f1b74104a039)
New Behavior:  On starting to drive, each tire in the TPMS GUI now shows a check mark if the tire sensor is confirmed working and tire pressure not very low.  Otherwise, that tire label comes up RED and with an Alert triangle.  
![Screenshot_20251118_171004_Open Vehicle](https://github.com/user-attachments/assets/55c3143d-0d01-4507-b50e-3a77690032cf)

After about a minute when first pressures come in, the triangles and check marks go away and are replaced by pressures.  
![Screenshot_20251118_171037_Google](https://github.com/user-attachments/assets/681acf07-7ee8-4470-9884-e5326d3512b4)

The color assignments remain as long as alert conditions persist.  Testing confirmed the RED alert goes away a few minutes after correcting a low pressure condition. 

Remaining issue: Left vs Right tire data always or often appears switched, however this is likely due to the Automatic TPMS Learning incorporated with THINK City cars.  Direct TPMS CAN data collection has shown that either the L/R tire order in the data is flipped or that it could go either way when the TPMS sensors are automatically "learned".  Until we know for sure which it is, no sense in changing anything.

zorgms commented:
Like this?

<img width="1392" height="785" alt="image" src="https://github.com/user-attachments/assets/eac2be5c-8e77-429d-a5e6-1efd2112c79f" />

dexterbg commented:
Myles, in case you missed this: ZorG already implemented the mapping for you in PR #1268. You only need to pull the latest `master` to get the mapping feature.

MylesTwete commented:
Thanks, yes, I just took the latest Edge build out for a spin with one of my cars.  I tested out the Tire Change buttons in both the TPMS tab in the OVMS UI and the TPMS change popup in the Android App.  I first swapped L/R in the OVMS UI and saved.  Testing on the road showed the Front tire orientation was now corrected but the Rear orientation was now wrong.  So in the App, I swapped the RR and LR.  The tire pressure orientation was now correct but the RR tire data completely disappeared on the screen.  Tapping on where the tire data should be, the L/R popup came up and I changed the orientation for that tire.  At first it seemed it was fixed, but then I noticed they were both showing the same tire pressure (actual should show 4psi diff).  So it seems there's a bug in this.  Still, overall, it's great to have this tool---and very timely for me.
![Screenshot_20251122_133323_Open Vehicle](https://github.com/user-attachments/assets/872e5557-1b1b-4cdf-9e12-621bc3b39f10) 
![Screenshot_20251122_133238_Open Vehicle](https://github.com/user-attachments/assets/1860a204-3d0d-4e1c-9e7c-3c4e8d6c10a3)
![Screenshot_20251122_133338_Open Vehicle](https://github.com/user-attachments/assets/efdae83f-d31d-4fbf-abf1-5b482900e700)

dexterbg commented:
Myles, please check/post the `tpms.*` part of your `config list vehicle` output. From your description I'd say both your rear sensors now map to the same wheel.

Also, it seems ZorG got the rear left/right assignment wrong, assuming the standard order left-right, while your sensor interpretation is right-left. I'd say you need to exchange the `m_tpms_index` indices 2 & 3 in both ID cases to fix that.

dexterbg commented:
→ https://github.com/openvehicles/Open-Vehicle-Monitoring-System-3/pull/1271#issuecomment-3568221613
